semgrep/semgrep/evaluation.py

```diff
@@ -207,16 +207,21 @@ def evaluate(
         }
     ]
     debug_print(str(pattern_ids_to_pattern_matches))
-    valid_ranges_to_output = evaluate_expression(
-        rule.expression,
-        pattern_ids_to_pattern_matches,
-        flags={RCE_RULE_FLAG: allow_exec},
-        steps_for_debugging=steps_for_debugging,
-    )
+    if rule._mode == "search":
```

Comment:
We probably want to be consistent with the `_mode` comparisons, so we have a consistent fallback mode.

Reply:
good point- just fixed this